src/components/TranslateBoxes.js

```diff
@@ -3,27 +3,62 @@ import Grid from '@material-ui/core/Grid';
 import Paper from '@material-ui/core/Paper';
 import TextField from '@material-ui/core/TextField';
 import Box from '@material-ui/core/Box';
+import { Typography, Tooltip, Link } from '@material-ui/core';
 
 
 export default function TranslateBoxes({input, handleInputChange, output, errors, classes}){
-    function addErrorsToOutput(output, errors){
-        if (errors && Object.entries(errors).length !== 0){
-            let finalOutput = output
 
-            // go through output text and find any strings with pattern $$E_$$
-            const regexp = /\$\$E.\$\$/g
-            const matches = [...output.matchAll(regexp)];
+    /**
+     * Function takes output and errors and returns a list of typography and tooltip
+     * elements to be displayed in the output area. Errors are found in the output and replaced
+     * with "<err>" as well as a tooltip explaining the error.
+     * @param {response translation from backend} output 
+     * @param {response errors from backend} errors 
+     */
+    function createOutputTypography(output, errors){
+        const regexp = /\$\$E.\$\$/g
```

Comment:
This looks like it would only match a singe char and `E404` would not be captured.
I would also make the regex a little more specific and make it look for digits instead of wildcard (ex. `E\d+`).